### PR TITLE
Ensure consistent type renaming of heap objects of dynamic size

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

File: src/ansi-c/c_typecheck_type.cpp [hunk 1]
```diff
@@ -581,10 +581,9 @@ void c_typecheck_baset::typecheck_array_type(array_typet &type)
       new_symbol.base_name=id2string(current_symbol.base_name)+suffix;
       new_symbol.type=size.type();
       new_symbol.type.set(ID_C_constant, true);
-      new_symbol.is_type=false;
-      new_symbol.is_static_lifetime=false;
       new_symbol.value=size;
       new_symbol.location=source_location;
+      new_symbol.mode = ID_C;
```
Comment (kroening) [inline src/ansi-c/c_typecheck_type.cpp]: ID_C should be 'mode' (a member of c_typecheck_baset)

File: src/goto-symex/symex_builtin_functions.cpp [hunk 2]
```diff
@@ -135,14 +135,15 @@ void goto_symext::symex_allocate(
...
-      symbolt size_symbol;
+      auxiliary_symbolt size_symbol;
 
       size_symbol.base_name=
         "dynamic_object_size"+std::to_string(dynamic_counter);
       size_symbol.name="symex_dynamic::"+id2string(size_symbol.base_name);
-      size_symbol.is_lvalue=true;
       size_symbol.type=tmp_size.type();
       size_symbol.mode=ID_C;
+      size_symbol.type.set(ID_C_constant, true);
```
Comment (kroening) [inline src/goto-symex/symex_builtin_functions.cpp]: Same as above, while at it!

**Review by kroening — APPROVED** [review]

File: src/ansi-c/c_typecheck_type.cpp (hunk 1 above)
Comment (tautschnig) [reply src/ansi-c/c_typecheck_type.cpp]: Will fix!

File: src/goto-symex/symex_builtin_functions.cpp (hunk 2 above)
Comment (tautschnig) [reply src/goto-symex/symex_builtin_functions.cpp]: I'm not sure what I should make the same as above? I can't use a member of `c_typecheck_baset` here?!

Comment (tautschnig) [reply src/goto-symex/symex_builtin_functions.cpp]: Sorry, I should be picking up the fix that @peterschrammel has provided and use `mode`, understood.

**Pull request merged**